yt_dlp/extractor/tvp.py

```diff
@@ -361,38 +390,51 @@ def _real_extract(self, url):
         # stripping JSONP padding
         datastr = webpage[15 + len(callback):-3]
         if datastr.startswith('null,'):
-            error = self._parse_json(datastr[5:], video_id)
-            raise ExtractorError(error[0]['desc'])
+            error = self._parse_json(datastr[5:], video_id, fatal=False)
+            error_desc = traverse_obj(error, (0, 'desc'))
+
+            if error_desc == 'Obiekt wymaga płatności':
+                raise ExtractorError('Video requires payment and log-in, but log-in is not implemented')
+
+            raise ExtractorError(error_desc or 'unexpected JSON error')
 
         content = self._parse_json(datastr, video_id)['content']
         info = content['info']
         is_live = try_get(info, lambda x: x['isLive'], bool)
 
+        if info.get('isGeoBlocked'):
+            # actual country list is not provided, we just assume it's always available in PL
+            self.raise_geo_restricted(countries=['PL'])
+
         formats = []
         for file in content['files']:
-            video_url = file.get('url')
+            video_url = url_or_none(file.get('url'))
             if not video_url:
                 continue
-            if video_url.endswith('.m3u8'):
+            ext = determine_ext(video_url)
+            if ext == 'm3u8':
                 formats.extend(self._extract_m3u8_formats(video_url, video_id, m3u8_id='hls', fatal=False, live=is_live))
-            elif video_url.endswith('.mpd'):
+            elif ext == 'mpd':
                 if is_live:
                     # doesn't work with either ffmpeg or native downloader
                     continue
                 formats.extend(self._extract_mpd_formats(video_url, video_id, mpd_id='dash', fatal=False))
-            elif video_url.endswith('.f4m'):
+            elif ext == 'f4m':
                 formats.extend(self._extract_f4m_formats(video_url, video_id, f4m_id='hds', fatal=False))
             elif video_url.endswith('.ism/manifest'):
                 formats.extend(self._extract_ism_formats(video_url, video_id, ism_id='mss', fatal=False))
             else:
                 # mp4, wmv or something
                 quality = file.get('quality', {})
+                # API started returning 0 in some formats???
+                if not isinstance(quality, dict):
+                    quality = {}
```

Comment:
```suggestion
                quality = file.get('quality') or {}
```

Reply:
I fear they might be returning numbers other than 0 too